### PR TITLE
reusable line chart / step 1

### PR DESCRIPTION
**Pull request by nitoloz**

### CODE REVIEW

File: src/charts/line/reusable-line-chart/index.stories.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        "date": "2018-12-31",
+        "quarter_label": "END Q3",
+        "financial_year_label": "",
+        "total_spent_to_date": null,
+        "total_spent_in_quarter": null,
+        "total_estimated_project_cost": 3650000,
+        "status": "Feasibility"
+    },
+    {
+        "date": "2018-03-31",
```
Comment (jbothma) [inline src/charts/line/reusable-line-chart/index.stories.js]:
```suggestion
        "date": "2019-03-31",
```

**Comment by nitoloz** [comment]
Comments from JD:
* Tooltips 
   * ~~it's possible for total project cost to be null - in those cases there shouldn't be a point or tooltip for it, it should just connect the line from the previous data point. if there's no previous data point (if the first total cost(s) is/are null) then it should just be a horizontal line up to the first point~~

   * ~~could you add a storybook showing how the tooltips work when the cumulative and total lines are very close to each other? e.g. one quarter cumulative is just under total, next quarter they're same, next quarter cumulative is just over? Somehow the tooltips should not overlap and still be readable - I think one solution is to look at which is lower when they would overlap, and that one should go down rather than up~~
   * ~~tooltips sometimes draw in the wrong place~~
   * ~~could you add a story for the null total cost issue as well?~~
   * ~~tooltip wrapping issue~~
* Axes
   * ~~could you add a story showing that it formats the y-axis ticks like R100k for hundreds of thousands and has enough space? R is Rands for South Africa's currency, you probably already know :)~~

   * ~~could you give the y-axis labels more space so they don't touch the ticks?~~

   * ~~could you add the ticks for the rest of the x- and y-axes - not just the top value?~~

   * ~~could you remove the ticks from the origin?~~
 * Other 
   *  ~~could you add the legend? I think you forgot that~~
   * ~~could you add a story with the chart 320 wide and 450 high with 5px page margin~~

File: src/charts/line/reusable-line-chart/reusable-line-chart.js [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+import {easeLinear} from 'd3-ease';
+import {max, min} from 'd3-array';
+import {select} from 'd3-selection';
+import {format} from 'd3-format';
+import {line} from 'd3-shape';
+import d3Tip from "d3-tip";
+
+const margin = {top: 50, right: 50, bottom: 80, left: 60};
+
+Object.defineProperty(Array.prototype, 'flat', {
```
Comment (jbothma) [inline src/charts/line/reusable-line-chart/reusable-line-chart.js]: Is this a polyfill? Could you feature test for it rather than blanket override it?

Comment (jbothma) [reply src/charts/line/reusable-line-chart/reusable-line-chart.js]: And and add a comment indicating that's what's happening here

Comment (nitoloz) [reply src/charts/line/reusable-line-chart/reusable-line-chart.js]: You're right, this isn't actually needed, so removed these lines

**Pull request merged**